install/kubernetes/helm/istio/templates/configmap.yaml

```diff
@@ -19,6 +19,20 @@ data:
     disablePolicyChecks: true
     {{- end }}
 
+    {{- if .Values.mixer.telemetry.reportBatchMaxEntries }}
+    # Set reportBatchMaxEntries to 0 to use the default batching behavior. A positive
+    # value indicates the number of requests that are batched before telemetry data 
+    # is sent to the mixer server
```

Comment:
the comment should be moved in values.yaml as we have tool to parse comments in values.yaml and put it out in istio.io

Reply:
The problem is that when the value is not defined, it just leaves these 3 lines of comments and nothing else. It's kind of confusing to look at. If you scroll down some lines, you can see other parameters' comments are enclosed within the conditional statement if there's a chance that the parameter is not defined.

Reply:
I'm confused why this is an issue.  That is what is in line 44 of mixer/values.yaml file too.

I just realized all the install related changes need to be in istio/installer as well, you can do that in a separate PR.

Reply:
> I'm confused why this is an issue. That is what is in line 44 of mixer/values.yaml file too.
So, if the comment is outside of the conditional statement, the generated configmap will look like:
```
    # Set the following variable to true to disable policy checks by the Mixer.
    # Note that metrics will still be reported to the Mixer.
    disablePolicyChecks: true

    # Set reportBatchMaxEntries to 0 to use the default batching behavior. A positive
    # value indicates the number of requests that are batched before telemetry data 
    # is sent to the mixer server

    # Set reportBatchMaxTime to 0 to use the default batching behavior. A positive time 
    # value indicates the maximum wait time since the last request will telemetry data 
    # be batched before being sent to the mixer server
```
To me, this is confusing as the comment exists but no parameters. If you look at the same file, but a few lines down, you will see things such as
```
    {{- if or .Values.mixer.policy.enabled (and .Values.global.istioRemote .Values.global.remotePolicyAddress) }}
    # policyCheckFailOpen allows traffic in cases when the mixer policy service cannot be reached.
    # Default is false which means the traffic is denied when the client is unable to connect to Mixer.
    policyCheckFailOpen: {{ .Values.global.policyCheckFailOpen }}
    {{- end }}
```
or
```
    {{- if .Values.gateways.enabled }}
    # Let Pilot give ingresses the public IP of the Istio ingressgateway
    ingressService: istio-ingressgateway
    {{- end }}
```
where comment is only there when a parameter is actually defined.

Reply:
The comment should be in values.yaml

Reply:
@linsun yeah, I read your comment too quickly. Didn't realize it until I see @howardjohn 's comment. Yeah makes sense to move the comment to values.yaml.

Reply:
OK, i made the comment in configmap.yaml more succinct, and the more description version of the default behavior is put into value.yaml per the above suggestions.